### ISSUE TITLE
Pyasn installed but opensnitch does not find

### ISSUE THREAD

**Issue by Ygarr** [evilsocket/opensnitch#795]
Cannot run opensnitch-ui

last version of opensnitch
OS: Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.5 LTS
Release:	20.04
Codename:	focal
 
Installed all packages

log in terminal:
```
Loading translations: /usr/lib/python3/dist-packages/opensnitch/i18n locale: en_US
using IPASN DB: /usr/lib/python3/dist-packages/data/ipasn_20140513_v12.dat.gz
exception loading ipasn db: [Errno 2] No such file or directory: '/usr/lib/python3/dist-packages/data/ipasn_20140513_v12.dat.gz'
Install python3-pyasn to display IP's network name.
```
but 
```
pip install  pyasn
Requirement already satisfied: pyasn in /usr/local/lib/python3.8/dist-packages (1.6.1)
```
and
```
python3-pyasn1/focal,focal,now 0.4.2-3build1 all [installed]
  ASN.1 library for Python (Python 3 module)

python3-pyasn1-modules/focal,focal,now 0.2.1-0.2build1 all [installed]
  Collection of protocols modules written in ASN.1 language (Python 3)
python-pyasn1/focal,focal,now 0.4.2-3build1 all [installed]
  ASN.1 library for Python (Python 2 module)
pypy-pyasn1/focal,focal,now 0.4.2-3build1 all [installed]
  ASN.1 library for Python (PyPy module)


```

**Comment by gustavo-iniguez-goya** [comment]
Hi @Ygarr ,

You're right, it seems that the package does not distribute the db /usr/lib/python3/dist-packages/data/ipasn_20140513_v12.dat.gz for some python versions. I'll investigate it.

For the time being, you can download the default db from their repository (https://github.com/hadiasghari/pyasn):
```bash
~ $ cd .config/opensnitch/
~ $ wget https://github.com/hadiasghari/pyasn/blob/master/data/ipasn_20140513_v12.dat.gz?raw=true -O ipasn_db.dat.gz
~ $ wget https://github.com/hadiasghari/pyasn/blob/master/data/asnames.json?raw=true
```

> python3-pyasn1

Regarding this package, it's for a different purpose. not for Adresses lookup.

**Comment by gustavo-iniguez-goya** [comment]
In order to download latest db:
```bash
~ $ cd .config/opensnitch
~ $ pyasn_util_download.py --latest # <- this will download the file rib.*
~ $ pyasn_util_convert.py --single ./rib.20230110.1400.bz2 ipasn_db.dat
~ $ gzip ipasn_db.dat
```

**Comment by grawlinson** [comment]
I stumbled upon this issue when looking into finding an ASN database, so here's how to obtain a recent-ish `asnames.json` file courtesy of [RIPE](https://www.ripe.net).

```sh
cd .config/opensnitch
curl -LO https://ftp.ripe.net/ripe/asnames/asn.txt
```

Then use the following Python script to convert the text file to `asnames.json`:

```python
#!/usr/bin/env python3
# SPDX-FileCopyrightText: 2023 George Rawlinson <george@rawlinson.net.nz>
# SPDX-License-Identifier: GFDL-1.3-or-later

import json

# initialise buffer for file contents
buf = {}

with open("asn.txt") as fh:
    for line in fh:
        # extract id + name
        asn_id, asn_name = line.split(" ", maxsplit=1)

        # add to buffer (minus trailing newline)
        buf[asn_id] = asn_name.rstrip()

# dump buffer to asnames.json
out_file = open("asnames.json", "w")
json.dump(buf, out_file)
out_file.close()
```

**Comment by grawlinson** [comment]
I’ve looked at the upstream scripts for pyasn as I’ve recently packaged pyasn for Arch Linux, the data directory isn’t supposed to be included when installed as it is only for examples & testing. So Debian is actually breaking upstream pyasn intention by erroneously including it. 

Not to mention that the data included is a very stale snapshot from 2015(?), so it would be a good idea if this code was refactored to reflect that.

**Comment by gustavo-iniguez-goya** [comment]
We should be able to update this list automatically with the downloader plugin: https://github.com/evilsocket/opensnitch/tree/master/ui/opensnitch/plugins/downloader

I've been using it to update the lists of malware/ads , but we can also use it for things like this.

**Comment by kuvaldini** [comment]
File: `download_asnames_json.py`
```python3
#!/usr/bin/python3
import json,sys,urllib.request as req; 
with open('asnames.json','w') as f: json.dump((dict(line.decode('utf-8','ignore').rstrip().split(maxsplit=1) for line in req.urlopen('https://ftp.r
ipe.net/ripe/asnames/asn.txt'))),f)
```
PRODUCES `asnames.json`

File: `download_pyasn.sh`
```bash
#!/bin/bash
set -euo pipefail
pyasn_util_download.py --latestv46 --filename ./ipasn.rib.bz2
pyasn_util_convert.py --compress --single ./ipasn.rib.bz2 ipasn_db.dat
rm ipasn.rib.bz2
```
PRODUCES `ipasn_db.dat.gz`

both scripts should be started from `~/.config/opensnitch`

**Comment by kuvaldini** [comment]
@gustavo-iniguez-goya how to use that plugin?

**Comment by gustavo-iniguez-goya** [comment]
@kuvaldini you need to install the package `python3-pyasn` (https://github.com/hadiasghari/pyasn)
Then it'll load automatically.

**Comment by kuvaldini** [comment]
it did not work automatically, so i posted that, as that helped on my setup.
pyasn is definitely installed.

**Comment by gustavo-iniguez-goya** [comment]
The problem is that the DB shipped with pyasn is not updated, so a lot of IP address won't resolve the network name.
At least the package on Debian leaves the db under `/usr/lib/python3/dist-packages/data/`, so if the package `python3-pyasn` is installed it should load it.

In your case it seems that the package is installed, the GUI loaded it, but as the DB was outdated it was not displaing the networks.

We need to create an easy way to update the DB, and let the users configure how often to update it.
Basically create a new plugin that imitates what the scripts do: https://github.com/evilsocket/opensnitch/tree/master/utils/scripts

**Comment by kuvaldini** [comment]
somehow on arch linux that folder is `/usr/lib/python3/site-packages/data/` site- instead of dist-, any py package is there, i am not sure if that causes a problem but i never seen any data downloaded by pyasn (old or new).

i looked at that files downloaded pyasn and don't understand why to create a special package to:
- parse file like `/etc/hosts` with so obvious format
- download hosts-like databases from regular public HTTP remotes
- gzip, ungzip 
- and json conversations

from my point of view that is almost redundant dependancy, isn't it?
